app/models/competition.rb

```diff
@@ -95,6 +95,8 @@ class Competition < ApplicationRecord
     restricted: 2,
   }, prefix: true
 
+  enum :cancellation_restrictions, [:no_restrictions, :restrict_accepted, :restrict_paid], prefix: :registration_cancel
```

Comment:
I don't think you need to include the word `restrict` in the values for an enum which is already called `restrictions`.

Reply:
What I found in practice is that calling `competition.{enum_value}` makes it unclear what `no_restrictions` means in that context - I'll see how it comes out by taking the "when can they cancel" approach